src/Jackett.Common/Definitions/nyaasi.yml

```
@@ -155,13 +159,16 @@ search:
     details:
       selector: td:nth-child(2) a:last-of-type
       attribute: href
-    download:
+    download_optional:
       selector: td:nth-child(3) a[href$=".torrent"]
       attribute: href
       optional: true
-    magnet:
+    magnet_optional:
       selector: td:nth-child(3) a[href^="magnet:?"]
       attribute: href
+      optional: true
+    download:
+      text: "{{ if and (.Config.prefer_magnet_links) (.Result.magnet_optional) }}{{ or (.Result.magnet_optional) (.Result.download_optional) }}{{ else }}{{ or (.Result.download_optional) (.Result.magnet_optional) }}{{ end }}"
```

If magnets can also be missing (which I don't think is the case):

```suggestion
      text: "{{ if and (.Config.prefer_magnet_links) (.Result.magnet_optional) }}{{ or (.Result.magnet_optional) (.Result.download_optional) }}{{ else }}{{ or (.Result.download_optional) (.Result.magnet_optional) }}{{ end }}"
    magnet:
      text: "{{ if .Config.prefer_magnet_links }}{{ else }}{{ .Result.magnet_optional }}{{ end }}"
      optional: true
```

If magnets are always present, revert `magnet_optional` to `magnet`, remove its `optional: true`:

```suggestion
      text: "{{ if .Config.prefer_magnet_links }}{{ else }}{{ .Result.download_optional }}{{ end }}"
      optional: true
```

---

Having a re-read of the initial issue I'm realising I misunderstood it. It's not about missing torrent and magnet links, but rather that torrents can cause naming issues, so users want to be able to be able to prefer magnets to torrents.

In that case, I think my second suggestion is more fitting. We should always provide `magnet` links, and if `.Config.prefer_magnet_links` is true then we should just remove `download` (torrent) links completely.

---

Are you wanting to rework this or should I just commit it directly?